fix: Update Overview page permissions (M2-6555)

JustinNusca opened the pull request:
### 📝 Description

🔗 [M2-6555](https://mindlogger.atlassian.net/browse/M2-6555): [FE][Overview] Applet Overview Page

This PR fixes an issue where the UI state was not being handled correctly on the Overview page, when the user viewing the Overview page was not either the owner or reviewer for the given applet.

### 📸 Screenshots

| Before | After |
|-|-|
| ![localhost_3000_dashboard_c81bee80-1da8-479f-9237-212a850e0c58_overview (1)](https://github.com/ChildMindInstitute/mindlogger-admin/assets/1670836/7e142c5f-960d-495f-83c8-338846983b41) | ![localhost_3000_dashboard_c81bee80-1da8-479f-9237-212a850e0c58_overview](https://github.com/ChildMindInstitute/mindlogger-admin/assets/1670836/3e089e3a-79ac-4af2-8a78-83562b9bc908) |


### 🪤 Peer Testing

1. View an **Applet → Overview** page as a user with `Manager`, `Owner`, `SuperAdmin`, or `Reviewer` roles.
2. Notice that you are able to view the page as expected.
3. View an **Applet → Overview** page as a user with any role not listed above.
4. Notice that you are not able to view the page, and the "You do not have permission" notice is shown.

[M2-6555]: https://mindlogger.atlassian.net/browse/M2-6555?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

JustinNusca committed:
Update `Overview` page to account for role permissions.

ChaconC reviewed (dismissed):
Tested various roles and permissions are working as expected

JustinNusca committed:
Add `dependencies` parameter to `usePermissions`.

- This allows the effect that the `asyncFn` is called within to be re-triggered when the provided dependencies change.
- This fixes the broken pagination on the Overview page.

ChaconC reviewed (approved):
Working correctly with the last update 👍
